include/seqan/index/find2_index_approx.h

```diff
@@ -567,43 +724,57 @@ inline void _optimalSearchScheme(TDelegate & delegate,
                                  std::array<OptimalSearch<nbrBlocks>, N> const & ss,
                                  TDistanceTag const & /**/)
 {
+    auto dummy = [](auto /*iter*/, uint32_t /*needleLeftPos*/, uint32_t /*needleRightPos*/, uint8_t /*errors*/,
+                    auto /*s*/, uint8_t const /*blockIndex*/)
+    {
+        return(false);
+    };
+    auto delegateDummy = [](Pair<uint16_t, uint32_t> const & /*pos*/, DnaString const & /*needle*/,
+                            uint8_t const /*errors*/)
+    {};
+
     for (auto & s : ss)
-        _optimalSearchScheme(delegate, it, needle, s, TDistanceTag());
+        _optimalSearchScheme(delegate, delegateDummy, dummy, it, needle, s, TDistanceTag());
 }
 
+
 // ----------------------------------------------------------------------------
 // Function find()
 // ----------------------------------------------------------------------------
 
 template <size_t minErrors, size_t maxErrors,
-          typename TDelegate,
+          typename TDelegate, typename TDelegateD, typename TCondition,
           typename TText, typename TIndexSpec,
           typename TChar, typename TStringSpec,
           typename TDistanceTag>
 inline void
 find(TDelegate & delegate,
+     TDelegateD & delegateDirect,
```

Comment:
I would suggest keeping the find() function interface and code without itvCondition, i.e. don't change them at all

I might have said something different last time :grimacing: but after reconsideration there is no reason to have a find-interface with itvCondition (as we don't document it). Only _optimalSearchScheme() should have an itvCondition.

Later we can improve the find() functions by pre-implementing conditions but the user shouldn't have to do it himself/herself